main/lsp/requests/document_formatting.cc

```diff
@@ -80,7 +81,9 @@ void DocumentFormattingTask::preprocess(LSPPreprocessor &preprocessor) {
         sourceView = sorbet::FileOps::read(path);
     }
 
-    if (!sourceView.empty()) {
+    // Don't format `__package.rb` files, since currently formatting them
+    // can potentially break some pay-server tooling
+    if (!sourceView.empty() && !absl::EndsWith(path, "__package.rb")) {
```

Comment:
Can we use the logic built into Sorbet for this?

```cpp
config.uri2FileRef(typechecker.state(), params->textDocument->uri).data(gs).isPackage()
```

Reply:
Alternatively, move the `isPackagePath` helper out of `Files.cc` into a static helper method on `core::File` (so that you can work with string_view instead of `core::FileRef`)